### PR TITLE
[AAE-5027] - Fix preselection for pagination - skipped selection/pres…

### PR DESCRIPTION
**Pull request by arditdomi**
…election

**Please check if the PR fulfills these requirements**

> - [X] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [X] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [X] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://alfresco.atlassian.net/browse/AAE-5027

**What is the new behaviour?**
In some cases not all the preselected nodes can be selected, for example: when the pagination does not allow a file to be shown and the node that we want to preselect is not part of the current rows. Once the pagination gets updated, (for example maxItems from 10 to 20), the preselected nodes that couldn't be selected in the first place should get selected now that they are visible in the document list. 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Review by mauriziovitale — APPROVED** [review]

**Review by eromano — COMMENTED** [review]
Do we still remove the selection when the user change page?

**Comment by arditdomi** [comment]
> Do we still remove the selection when the user change page?

Yes we do

**Review by eromano — CHANGES_REQUESTED** [review]
Nope , the document list has to be agnostic of what kind of pagination is used

**Review by mauriziovitale — APPROVED** [review]

File: lib/content-services/src/lib/document-list/components/document-list.component.ts [hunk 1]
```diff
@@ -894,7 +896,7 @@ export class DocumentListComponent implements OnInit, OnChanges, OnDestroy, Afte
         this._pagination.maxItems = requestPaginationModel.maxItems;
         this._pagination.merge = requestPaginationModel.merge;
         this._pagination.skipCount = requestPaginationModel.skipCount;
-        this.reload();
+        this._pagination.merge ? this.reloadWithoutResettingSelection() : this.reload();
```
Comment (eromano) [inline lib/content-services/src/lib/document-list/components/document-list.component.ts:899]: reload() {
        this.ngZone.run(() => {
            this.resetSelection();
              if(this._pagination.merge){
                          this.reloadWithoutResettingSelection();
              }
        });
    }

Comment (eromano) [reply lib/content-services/src/lib/document-list/components/document-list.component.ts:899]: to me looks better leave all thsi responsibility in the reload method

File: lib/content-services/src/lib/document-list/data/share-datatable-adapter.ts [hunk 2]
```diff
@@ -279,15 +279,7 @@ export class ShareDataTableAdapter implements DataTableAdapter {
         }
 
         if (merge) {
-            const listPrunedDuplicate = shareDataRows.filter((elementToFilter: any) => {
```
Comment (eromano) [inline lib/content-services/src/lib/document-list/data/share-datatable-adapter.ts:282]: why you want to remove this logic?

**Review by eromano — CHANGES_REQUESTED** [review]

**Comment by arditdomi** [comment]
Closing it for now. It is very hard to implement a fix and it impacts a lot of areas.

**Pull request closed without merging**